roq-core/src/main/java/org/roqmessaging/core/Monitor.java

```diff
@@ -363,6 +375,17 @@ public void run() {
 				}finally {
 				}
 			}
+			// Write Heartbeat
+			if ((Time.currentTimeMillis() - lastHb) >= hbPeriod) {
+				try {
+					current = Time.currentTimeSecs();
+					logger.info("Writing hb " + basePort + " " + current);
+					localState.put("HB", current);
+					lastHb = Time.currentTimeMillis();
+				} catch (IOException e) {
+					logger.info("Failed to write in local db: " + e);
+				}
+			}
```

Comment:
This code is repeated in the Exchange, Monitor and ScalingProcess. It should probably be just a function call.